### DISCUSSION TITLE
May I ask about copilot security?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#69743] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have applied to copilot cli.

When you enter confidential information such as passwords in terminal with copilot cli, is there a possibility that the confidential information is transmitted to external parties or learned by AI?

**Comment by Akash1134** [comment, marked as answer]
GitHub Copilot transmits data to GitHub’s Azure tenant to generate suggestions, including both contextual data about the code and file being edited (“prompts”) and data about the user’s actions (“user engagement data”). Copilot applies several measures of protection to this data, including:

*   The transmitted data is encrypted both in transit and at rest; Copilot-related data is encrypted in transit using transport layer security (TLS), and for any data we retain at rest using Microsoft Azure’s data encryption (FIPS Publication 140-2 standards).

*   Prompts only persist long enough to return a response from GitHub’s Azure tenant and are then immediately discarded. Prompts are never stored; all processing is ephemeral and takes place in memory.

*   Where user engagement data is retained, access is strictly controlled. The data can only be accessed by (1) named GitHub personnel working on the Copilot team or on the GitHub platform health team, and (2) Microsoft personnel working on or with the Copilot team.

*   Role-based access controls and multi-factor authentication are required for personnel accessing user engagement data.

*   User engagement data is currently stored for 24 months.

**When transferring data to third party systems,** GitHub takes reasonable technical, operational and contractual measures necessary to protect customer data from unauthorized access, alteration, or destruction; maintain data accuracy; and help ensure the appropriate use of your personal data.

For more details and questions, please refer [Copilot Trust Centre](https://resources.github.com/copilot-trust-center/)

Thanks!

**Reply by ghost** [reply]
Thank you for your response.

Let me ask two additional questions.

- Am I right in thinking that passwords and other confidential information are transmitted as raw data on TLS?For example, if a vulnerability is found in TLS, is there a risk of leakage?
- I understood that copilot for CLI does not store data. copilot for CLI is not trained by AI as well as copilot for business, is that correct?

**Reply by kuroodo** [reply]
Keep in mind Copilot is not SOC 2 compliant. You'll have to rely on Microsofts own "trust me bro" certification. There are copilot alternatives, such as Codeium, that are SOC 2 comliant.
